### PR TITLE
Download OpenAPI JSON

### PR DESCRIPTION
**Pull request by jwdotjs**
# Todos

- [ ] Auth Headers
- [ ] Move and style Download OpenAPI JSON button to under config
- [ ] Ability to import markdown documentation?
- [ ] Better typespecs
- [ ] Unit Tests
- [ ] Update Changelog and README.md with examples

### CODE REVIEW

File: src/lib/helpers.tsx [hunk 1]
```diff
@@ -12,10 +12,31 @@ import {
...
+
+    const openApiJson = JSON.stringify(openApiResult);
+
+    const type = "application/json";
+    const downloadReportUrl = window.URL.createObjectURL(
+      new Blob([openApiJson], { type })
+    );
+
+    const aLink = document.createElement("a");
+    aLink.download = `${workspace.name.toLowerCase()}-openapi.json`;
```
Comment (matthew-kerle) [inline src/lib/helpers.tsx:33]:
```suggestion
    aLink.download = `smartrent-${workspace.name.toLowerCase()}-openapi.json`;
```

File: src/lib/openapi.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        type: getParameterType(property),
+      };
+
+      return memo;
+    },
+    {}
+  );
+}
+
+function getPayloadName(name: string) {
```
Comment (matthew-kerle) [inline src/lib/openapi.ts:45]: Totally up to you, but you could make some of these really small if they were arrow functions.

File: src/lib/openapi.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  return `${name.replace(" ", "")}Payload`;
+}
+
+function getObjectRequiredProperties(params: Param[]): string[] {
+  if (!params?.length) {
+    return [];
+  }
+  return compact(
```
Comment (matthew-kerle) [inline src/lib/openapi.ts:53]: Nit:
```suggestion
  }

  return compact(
```

**Review by matthew-kerle — COMMENTED** [review]
Some super minor nits...will review more thoroughly in a bit.

File: src/lib/helpers.tsx (hunk 1 above)
Comment (jwdotjs) [reply src/lib/helpers.tsx:33]: This is open source and can be installed by anyone, I don't think we want this namespace

**Pull request closed without merging**